guest/handle/api/handle.go

```diff
@@ -0,0 +1,52 @@
+/*
+   Copyright 2023 The Kubernetes Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package api
+
+import (
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/api"
+	internalproto "sigs.k8s.io/kube-scheduler-wasm-extension/guest/internal/proto"
+	klogapi "sigs.k8s.io/kube-scheduler-wasm-extension/guest/klog/api"
+)
+
+// Handle provides data and some tools that plugins can use.
+//
+// This contains functions like framework.Handle.
+type Handle interface {
```

Comment:
In wasm I don't think we need to introduce the handle type, which is similar to a New function. It may be better to not introduce that vocab and instead only define EventRecorder similar to KLog here. In other words I think probably delete handle to flatten the types.

Also, you want to be sure that the api can be useable without depending on implementation,and tests here and in the advanced example will prove that. See. https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension/blob/main/guest/klog/api/klog.go https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension/blob/main/guest/klog/api/klog_test.go and try to follow the style as it may prevent some pitfalls.

Reply:
> In wasm I don't think we need to introduce the handle type, which is similar to a New function. It may be better to not introduce that vocab and instead only define EventRecorder similar to KLog here. In other words I think probably delete handle to flatten the types.

@sanposhiho,
What do you think about this suggestion?
It seems reasonable as the handle on the guest side is more of a formality, so it's possible to call EventRecorder directly without the handle.
However, if we make this change, the pluginFactory will become unnecessary, which means significant changes to the current implementation of the PR.
I prefer current implementation because it's similar to current kubernetes implementation itself.

Reply:
I know you didn't ask me, but I think there is also a RATIONALE.md here which mentions for simplicitly we don't follow exactly upstream SDK, it also improves the performance of the plugin to not have layers. Following the prior patterns of simple binding of types needed defined here, is a good way to be consistent forward. If you want to go back and make this for the sake of it same as the main SDK, maybe do it everywhere so that  it isn't inconsistent. However, the problems solved will be re-introduced, so update the RATIONALE to say why you chose to revert.

Reply:
TL'DR' is study existing practice and the RATIONALE files which took a long time to work out. If you disagree, then you can change, just make sure you make the whole thing consistent with that change. The RATIONALE was made so that I could step away and lessons remain.

Reply:
> simplicitly we don't follow exactly upstream SDK, it also improves the performance and testability of the plugin to not have layers. 

Based on this merit, I thumb up for the suggestion from @codefromthecrypt, for now.
@Gekko0114 's point does also make sense though, we can consider taking such steps that make wasm SDK similar to the original scheduler SDK **later**, along with all the other interfaces + combatting problems introduced by them like @codefromthecrypt suggested, because it takes additional effort for us while being less important.

Reply:
Sure, thanks @sanposhiho  and @codefromthecrypt for your suggestions.
Removed handle and added tests for eventrecorder.